org.eclipse.jdt.core/model/org/eclipse/jdt/core/compiler/CompilationParticipant.java

```diff
@@ -143,6 +152,48 @@ public void processAnnotations(BuildContext[] files) {
 	// do nothing by default
 }
 
+/**
+ * Returns whether this participant is interested in post processing of generated bytecode.
+ * <p>
+ * Returning <code>true</code> enables the callback {@link #postProcess(BuildContext,ByteArrayInputStream)}, where this
+ * participant can influence produced class file bytes.
+ * </p>
+ * <p>
+ * Default is to return <code>false</code>.
+ * </p>
+ *
+ * @return whether this participant is interested in post processing
+ * @since 3.34
+ */
+public boolean isPostProcessor() {
+	return false;
+}
+
+/**
+ * Notifies this participant about a finished class compilation. This notification occurs right before the compiled
+ * class is written out to disk. This allows to perform arbitrary byte code manipulation on the generated class.
```

Comment:
Please add `Only sent to participants that answer true to {@link #isPostProcessor()}.`